test/parallel/test-worker-cleanexit-with-moduleload.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const { spawn } = require('child_process');
+
+// Harden the thread interactions on the exit path.
+// Ensure workers are able to bail out safe at
+// arbitrary execution points.
+
+if (process.argv[2] === 'child') {
+  const { Worker, isMainThread } = require('worker_threads');
+  if (isMainThread) {
+    for (let i = 0; i < 10; i++) {
+      new Worker("const modules = ['fs', 'assert', 'async_hooks'," +
+        "'buffer', 'child_process', 'net', 'http', 'https', 'os'," +
+        "'path', 'v8', 'vm'];" +
+        'modules.forEach((module) => {' +
+        'const m = require(module);' +
+        '});', { eval: true });
+
+      // Allow workers to go live.
+      setTimeout(() => {
+        process.exit(0);
```

Comment:
Would it make more sense to not spawn a child process and instead use `worker.terminate()` here? I think the test would test the same things, just without the redirection of a child process?

Reply:
@addaleax - I resorted to abrupt exit as:
 - I believe `worker.terminate` will make worker exit more systematic and organized, and mask what I would like to test: ability to come out clean if we stop the process at arbitrary execution points
 - I believe `worker.terminate` sequences are already present in many other worker based tests

do you find any drawback with the child process (and process exit) based approach?

Reply:
> I believe `worker.terminate` will make worker exit more systematic and organized, and mask what I would like to test: ability to come out clean if we stop the process at arbitrary execution points

Can you explain why you believe that? Both calling `process.exit()` in a parent thread and calling `worker.terminate()` stop the thread the same way, namely through calling `Worker::Exit()` from the parent thread.

> I believe `worker.terminate` sequences are already present in many other worker based tests

Yes, that’s true.

> do you find any drawback with the child process (and process exit) based approach?

I find it a bit more inconvenient to debug these tests, because of the extra process involved , because  you can’t run any meaningful diagnostics on the test itself, you have to figure out how to run the “actual” test.

Reply:
@addaleax - one more reason I forgot to list earlier. This test produced a number of unique failures prior to #25061 . And comparing with worker's existing test cases and reasoning about why this was catching some errors while those were not, my assessment was that it is due to the `process.exit` that differentiated this one, of course I did not check the code when I thought so.

Reply:
Yeah, I guess we could keep it as a regression test for that.

We still don’t need the child process, do we? (If we do: Please make sure that stderr + stdout are forwarded in case of a failure)

Reply:
@addaleax - pushed a change to that effect; ptal.